### DISCUSSION TITLE
Multiline wrapping issues

### DISCUSSION THREAD

**Discussion by LYU4662** [emilk/egui#2683] [Q&A]
I need multiple lines of fixed lines, and after using the number of lines to be displayed, the content in the multi-line moves up after the carriage is entered, the cursor is still on the last line, and the content in the multi-line can be viewed by sliding up and down through the scroll wheel.
  egui::TextEdit::multiline(&mut self.my_string).desired_rows(4)

I didn't try it out in the existing multi-line, hoping to have a friend guide me

This is the initial interface
![微信截图_20230206163301](https://user-images.githubusercontent.com/98568222/216923090-23e307bd-c20c-4b39-a1d4-948e1eea6677.png)

This is the interface after entering a few carriage returns
![微信截图_20230206163325](https://user-images.githubusercontent.com/98568222/216923094-58e7cf7e-6975-4ebc-998f-99489c0a91fc.png)

**Comment by emilk** [comment]
You want it to scroll? Put it in a `ScrollArea`

**Reply by LYU4662** [reply]
Thanks for the reply, I need scrollable areas and editable regions, equivalent to multiple lines plus multiple lines of text scrollable

**Reply by LYU4662** [reply]
It seems that I tried this method of yours before, but I will verify again if it works, thanks again

**Comment by LYU4662** [comment]
What I want to do is  this effect


https://user-images.githubusercontent.com/98568222/216938029-b07ae9e8-2e5d-4aa0-a700-0f034e477482.mp4


egui::ScrollArea::vertical().show(ui, |ui| {
                    let send_window = ui.add_sized(
                        [356.0, 80.0],
                        egui::TextEdit::multiline(&mut self.my_string),
                    );
                    if send_window.changed() {
                        // …
                        println!("send_window.changed");
                    }
                    if send_window.lost_focus() && ui.input().key_pressed(egui::Key::Enter) {
                        // …
                        println!("send_window.lost_focus");
                    }
                });

This is the effect when I use the scroll area

https://user-images.githubusercontent.com/98568222/216937909-4a30cd2c-7efb-4c17-a86f-3017b4ebceeb.mp4

**Reply by LYU4662** [reply]
There will be problems with the preview, and you can view it after downloading
